src/test/java/org/openrewrite/java/testing/assertj/JUnitAssertNotNullToAssertThatTest.java

```diff
@@ -246,8 +246,9 @@ private String notification() {
     @Test
     @Issue("https://github.com/openrewrite/rewrite-testing-frameworks/issues/491")
     void importAddedForCustomArguments() {
+        //language=java
         rewriteRun(
-          //language=java
+          spec -> spec.typeValidationOptions(TypeValidation.none()),
```

Comment:
As it seems like this test passed the type validation before, I wonder why it doesn't anymore.

Reply:
That test links to this issue, which was the reason to add contextSensitive as a quick solution at the time, with unfortunate runtime performance impact.
- https://github.com/openrewrite/rewrite-testing-frameworks/issues/491

Those custom classes are not available when we drop `.contextSensitive()`, and the missing types would then break chained recipe runs to for instance adopt more optimal assertion patterns with AssertJ, as part of the migration.

Not quite sure where that leaves this effort. For a one time migration I'd lean towards taking the performance penalty for a more complete / idiomatic migration, but I don't know what prompted the need to fix the performance issue here. Any context there @philippe-granet ?

Reply:
I see. The template parameter's type is defined on the source path. That is a rather special case of course. So the test would also fail for a context-sensitive recipe, if the `A` class were defined in another source file.

I am not sure, but I would probably merge the PR and live with this limitation for now.

Thoughts?

Reply:
Ok;  I'll trust your judgement on that then and get this merged as is. Thanks a lot both!